Mathlib/Tactic/Linter/Multigoal.lean

```diff
@@ -0,0 +1,158 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+#  The "multiGoal" linter
+
+The "multiGoal" linter emits a warning where there is more than a single goal in scope.
+
+There are a few tactics that are intended to work specifically in such a situation and the linter
+ignores them. Otherwise, whenever a tactic leaves multiple goals, the linter will emit a warning,
+unless some form of "focusing" tactic is used.
+Typically, the focusing is achieved by the `cdot`: `·`, but, e.g., `focus` or `on_goal x` also
+serve a similar purpose.
+
+TODO:
+* Should the linter flag unnecessary scoping as well?
+  For instance, should
+  ```lean
+  example : True := by
+    · · exact .intro
+  ```
+  raise a warning?
+* Custom support for "accumulating side-goals", so that once they are all in scope
+  they can be solved in bulk via `all_goals` or a similar tactic.
+* In development, `on_goal` has been partly used as a way of silencing the linter
```

Comment:
Is this item still current? My understanding is that nobody complained much about this.